extension/src/popup/views/RecoverAccount/index.tsx

```diff
@@ -71,15 +125,32 @@ export const RecoverAccount = () => {
     }
   }, [publicKey]);
 
+  useEffect(() => {
+    const phraseInputsArr = [];
+
+    // eslint-disable-next-line no-plusplus
+    for (let i = 1; i <= maxPhraseLength; i++) {
+      phraseInputsArr.push(`MnemonicPhrase-${i}`);
+    }
+    setPhraseInputs(phraseInputsArr);
+  }, [maxPhraseLength]);
+
+  const handleMnemonicInputChange = (value: string, i: number) => {
+    const arr = [...mnemonicPhraseArr];
+    arr[i] = value;
+
+    setMnemonicPhraseArr(arr);
```

Comment:
as the user makes changes to a phrase input, construct the mnemonic phrase in memory by placing the input in the correct index of the string